holoviews/element/raster.py

```diff
@@ -279,6 +279,11 @@ def __init__(self, data, kdims=None, vdims=None, bounds=None, extents=None,
         l, b, r, t = bounds.lbrt()
         xdensity = xdensity if xdensity else compute_density(l, r, dim1, self._time_unit)
         ydensity = ydensity if ydensity else compute_density(b, t, dim2, self._time_unit)
+        if not np.isfinite(xdensity) or not np.isfinite(ydensity):
+            raise ValueError('Density along Image axes could not be determined. '
+                             'If the data contains only one coordinate along the '
+                             'x- or y-axis ensure you declare the bounds and/or '
+                             'density.')
```

Comment:
And compute_density will really return a floating-point inf value, not raise an exception?